### ISSUE TITLE
Error while saving

### ISSUE THREAD

**Issue by khinsen** [xmaysonnave/tiddlywiki-ipfs#1]
First of all, thanks for this very promising work!

I added the IPFS pluging to a new TiddlyWiki, and tried to create a new tiddler. When I pressed "save", I got the error message:
```
Error while saving:

Error: Ipfs default client is unavailable: Invalid scope
```

I get the same error message no matter if IPFS Desktop is running on my machine or not (normally it's running, but I stopped it for testing).

However, I do not get that error message when I open https://tiddly.bluelightav.eth.link/#%24%3A%2Fplugins%2Fipfs and create a new tiddler right there. However, if I press the "export to IPFS" button on the freshly created tiddler, I get the error message
```
Error while saving:

Upload to Ipfs not supported: text/html
```

**Comment by xmaysonnave** [comment]
Thanks for your report.
Here are some remarks :
1 - The default behaviour is to use the HTTP client who uses the infura.io api url and the gateway.ipfs.io gateway url. It means that you could upload your tiddly to Ipfs from Infura.
2 - It sounds to me that Ipfs Desktop should work as long as you install Ipfs Companion in your browser. In that use case. Even this plugin is setup to talk to infura.io the Ipfs Companion extension will catch the request and redirect on your local node (Ipfs Desktop). Feedback will be greatly appreciated as Ipfs Desktop do not run on my Debian 10 buster. I will try to investigate this behaviouron another platform (W10).
3 - Only file attachments are supported to be uploaded. Here is an exhaustive list of content types supported :

**Comment by xmaysonnave** [comment]
I made some tests with Ipfs Desktop along Ipfs Companion. 
Its is working as described earlier.
The default api and gateway url are redirected to your Ipfs desktop node.
if you choose https://tiddly.bluelightav.eth.link, Ipfs Companion redirects correctly to an ipns url. 
The version I uploaded has a preliminary ens support (Metamask browser extension).
Ipfs Companion and Metamask do not live well together. Metamask is able to redirect your .eth TLD to the proper ipns url however sometimes you can experience some freeze if Ipfs Companion is also active. Firefox warned me as both Ipfs Companion and Metamask bundle the web3.js library who manages ENS.
Thanks

**Comment by khinsen** [comment]
Thanks for your detailed reply! In summary, the first error I reported is due to Metamask, and the second one due to me trying an operation that is not supported (exporting something other than an attachment).

I tried exporting attachments using https://tiddly.bluelightav.eth.link/. It works fine with or without IPFS Desktop active. No noticeable speed difference, but I only tried small images. BTW, I didn't know that infura.io provides a gateway that allows storing data.
